pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2242,12 +2252,11 @@ func buildListener(opts buildListenerOpts) *xdsapi.Listener {
 	return listener
 }
 
-// appendListenerFallthroughRoute adds a filter that will match all traffic and direct to the
-// PassthroughCluster. This should be appended as the final filter or it will mask the others.
-// This allows external https traffic, even when port the port (usually 443) is in use by another service.
-func appendListenerFallthroughRoute(l *xdsapi.Listener, opts *buildListenerOpts,
-	node *model.Proxy, currentListenerEntry *outboundListenerEntry) {
-	wildcardMatch := &listener.FilterChainMatch{}
+// Create pass through filters for the listener assuming all the other filter chains are ready.
+// The match member of pass through filter chain depends on the existing non-passthrough filter chain.
+// TODO: Calculate the filter chain match to replace the wildcard and replace appendListenerFallthroughRoute.
```

Comment:
TODO(lambdai)

Reply:
Good catch! This TODO is fixed This is the legacy comment of the previous #23071 
Once that PR is merged I will merge and clean this.

This PR acts as the motivation of #23071 I am sorry I didn't find the explicit way of warning this PR contains #23071